### DISCUSSION TITLE
Como mudar o tipo de uma variável que usar função leia para numeroInteiro?

### DISCUSSION THREAD

**Discussion by BlackCattowo** [DesignLiquido/delegua#305] [P&R] [answered]
![duvida-delegua](https://user-images.githubusercontent.com/128187479/228991394-6249b5c5-7561-40c1-bfc1-605eef3ec5bc.png)

Eu estou aprendendo a programar ainda, sei um pouco de Python e fiz um programa similar em Python (acho que está disponível no meu perfil), para aprender e praticar Delegua eu normalmente converto programas que fiz em Python para Delegua, porém com esse programa em especifico eu não estou conseguindo fazer isso por não saber como fazer uma variável baseada no input do usuário em que o valor seja lido como um numero inteiro.

Será que alguém mais experiente poderia me ajudar?
Agradeço desde já por qualquer ajuda!

**Comment by leonelsanchesdasilva** [comment, marked as answer]
Olá, @BlackCattowo. Bem-vindo!

Basta usar:

`var n = inteiro(leia("Quantidade de números a serem analisados"))`

`leia()` por padrão devolve um texto. `inteiro()` faz parte da biblioteca padrão de Delégua.

**Reply by BlackCattowo** [reply]
Muito obrigado! Vou testar aqui.

**Reply by BlackCattowo** [reply]
Bom, aparentemente isso resolveu o primeiro problema do código, mas após eu rodar ele, ocorreu algum erro no console. Pode me ajudar a identificar ele e o corrigir? (Não queria abrir outra questão na página de discussão, por isso estou usando essa mesma postagem, mas se for melhor pra você abrir uma questão para cada dúvida eu farei isso na próxima ver.)

Desde já agradeço novamente!
![duvida delegua 2](https://user-images.githubusercontent.com/128187479/228997477-8b08fe2e-0505-43f4-b350-b1025f1cc5b8.png)

**Reply by samuelrvg** [reply]
Olá @BlackCattowo, nesse caso você esta usando o padrão do **python** com identação de espaços, mas precisa adicionar chaves `{}` sobre o bloco `enquanto` e sobre o condicional `se` exemplo:


```
var n = inteiro(leia('numeros a serem analisados: '))

var total = 0
var par = 0 

enquanto(n > 0){
    total += 1
    var x = inteiro(leia())
    se (x % 2 == 0){
        par += 1
    }
    n -= 1
}

var impar = total - par
escreva(par, 'pares')
escreva(impar, 'impares')
```

Temos também o dialeto **EguaP** para escrever de forma identada como `python` veja mais sobre [clicando aqui](https://github.com/DesignLiquido/delegua/wiki/Dialetos#eguap)

**Reply by BlackCattowo** [reply]
Opa. Muito obrigado, o programa deu certo. Quanto ao EguaP eu irei checar depois, mas provavelmente continuarei usando o Delegua porque a ideia de uso dessa lingua é justamente para eu aprender algo diferente do Python. Mas obrigado pela ajuda e recomendação!
